### PR TITLE
Fix hardcoded API base 

### LINKED ISSUE #9013: Custom Api Base not working in Openai Handler
### Discussed in https://github.com/mindsdb/mindsdb/discussions/9010

<div type='discussions-op-text'>

<sup>Originally posted by **ArnavK-09** March 30, 2024</sup>
Using the ` api_base ` option..</div>

### PR DESCRIPTION
**Pull request by ZoranPandovski**
## Description

This PR removes the hardcoded base URL for openai and also provides an option to read the base URL from connection args created with CREATE ML Engine

Fixes #9013

## Type of change

(Please delete options that are not relevant)

- [X] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines(PEP 8) of MindsDB.
- [X] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/openai_handler/openai_handler.py [hunk 1]
```diff
@@ -201,11 +204,8 @@ def predict(self, df: pd.DataFrame, args: Optional[Dict] = None) -> pd.DataFrame
 
         pred_args = args['predict_params'] if args else {}
         args = self.model_storage.json_get('args')
-        args['api_base'] = pred_args.get(
-                    'api_base',
-                    args.get(
-                        'api_base', os.environ.get('OPENAI_API_BASE', OPENAI_API_BASE)
-                    ))
+        connection_args = self.engine_storage.get_connection_args()
+        args['api_base'] = connection_args.get('api_base') or args.get('api_base') or os.environ.get('OPENAI_API_BASE', OPENAI_API_BASE)
```
Comment (paxcema) [inline mindsdb/integrations/handlers/openai_handler/openai_handler.py:208]: I think this is missing `or pred_args.get('api_base')`

File: mindsdb/integrations/handlers/openai_handler/openai_handler.py [hunk 2]
```diff
@@ -693,8 +691,6 @@ def finetune(
         api_key = get_api_key(self.name, args, self.engine_storage)
 
         using_args = args.pop('using') if 'using' in args else {}
-        prompt_col = using_args.get('prompt_column', 'prompt')
```
Comment (paxcema) [inline mindsdb/integrations/handlers/openai_handler/openai_handler.py:696]: Curious as to why these were removed, are they not used?

**Review by paxcema — CHANGES_REQUESTED** [review]
Couple minor points I'd like addressed before approving, but it generally looks great (I especially like the change from nested `.get()`s to `or` 👌

File: mindsdb/integrations/handlers/openai_handler/openai_handler.py (hunk 1 above)
Comment (ZoranPandovski) [reply mindsdb/integrations/handlers/openai_handler/openai_handler.py:208]: Yes, adding it

File: mindsdb/integrations/handlers/openai_handler/openai_handler.py (hunk 2 above)
Comment (ZoranPandovski) [reply mindsdb/integrations/handlers/openai_handler/openai_handler.py:696]: Yes, prompt_col and completion_col were not referenced anywhere

**Comment by ZoranPandovski** [comment]
@paxcema Thanks for the review. I've added the required changes for `pred_args`.

**Review by paxcema — APPROVED** [review]

**Pull request merged**